dev-client/src/context/connectivity/connectivityLogic.ts

```diff
@@ -0,0 +1,37 @@
+/*
+ * Copyright © 2024 Technology Matters
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU Affero General Public License as published
+ * by the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the
+ * GNU Affero General Public License for more details.
+ *
+ * You should have received a copy of the GNU Affero General Public License
+ * along with this program. If not, see https://www.gnu.org/licenses/.
+ */
+
+const getIsOnline = ({
```

Comment:
this `getIsOnLine` seems unnecessary? unless you plan to use it in your `ConnectivityContext` later on it could be part of the `getIsOffline` function:
```
export const getIsOffline = ({
  isConnected,
  isInternetReachable,
}: {
  isConnected: boolean | null;
  isInternetReachable: boolean | null;
}) => {
  return !isConnected || !isInternetReachable;
};
```

Reply:
I mostly just had it for enforcement around when `getIsOffline` should return `null`, and that code has slightly different behavior when `isConnected` is true but `isInternetReachable` is null. I'm not sure how often we'll be in that state or how important it is to track, but since netinfo distinguishes between true / false / null, I figured we should too.